### ISSUE TITLE
aria-valid-attr-value should permit null values for attributes

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3489]
Code like the following should not fail the `aria-valid-attr-value` rule:

```html
<div role="checkbox" aria-checked>Accept terms and conditions</div>
```

We could consider reporting this as incomplete, possibly. This is sort of an odd thing to happen, but it isn't a guaranteed failure. `aria-checked` is not a required attribute for `checkbox`, so this is sort of like not having the attribute there at all.

I'm not sure this holds for all ARIA attributes, we'll need to check that. And I think we'll need to confirm that aria-required-attr isn't passed with null attributes. It seems to me that if we're going to fail null attributes its only for attributes that are required.

**Comment by WilcoFiers** [comment]
WAI-ARIA 1.2 changes this a bit. checkbox no longer has a default aria-checked value.

I did some testing
|           | NVDA+FF     | JAWS+GC     | VO+Safari   |
|-----------|-------------|-------------|-------------|
| no attr   | not checked | not checked | Not checked |
| null attr | not checked | not checked | Not checked |
| ""        | not checked | not checked | Not checked |
| "false"   | not checked | not checked | Not checked |
| "FALSE"   | checked     | not checked | Not checked |
| " false " | checked     | checked     | checked     |
| "invalid" | checked     | checked     | checked     |

On Google Chrome, NVDA and Narrator behave the same as JAWS.


I think how we address this issue depends on how we deal with #3461. If we change this rule to be a best practice I think it's okay to be stricter. If not, we could potentially just flag everything as "needs review". The way browsers have implemented this attribute is pretty smart.

Something like the following is will work consistently, even though both states are trigged through invalid values:

```jsx
const MyCheckbox = (props) => (<div role="checkbox" aria-checked={props.checked ? 'checked' : ''} />
```

**Comment by WilcoFiers** [comment]
We're going with the less strict option.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base,
Test script:
```
<div role="checkbox" aria-checked>Accept terms and conditions</div>
<div aria-invalid></div>
<div aria-hidden></div>
<div aria-checked=""></div>
```
And `aria-valid-attr-value` rule is part of incomplete on empty and null attributes.


![Image](https://user-images.githubusercontent.com/47409579/190192653-203cfc39-ff73-42f4-9ca5-8380b8f8f279.png)
